features/org.eclipse.equinox.executable.feature/library/eclipse.c

```diff
@@ -445,6 +445,20 @@ JNIEXPORT int run(int argc, _TCHAR* argv[], _TCHAR* vmArgs[])
 	setenv(firstThreadEnvVariable, "1", 1);
 #endif
 
+#ifdef LINUX
+	if (!noSplash) {
+		char *swtGtk4 = getenv("SWT_GTK4");
+		if (swtGtk4 != NULL && strcmp(swtGtk4, "1") == 0) {
+			// The eclipse launcher uses GTK3 exclusively, therefore when starting GTK4
+			// it must be started without a splash screen
+			// https://github.com/eclipse-equinox/equinox/issues/831
+			// If the user didn't already specify -nosplash print a warning and turn the splashscreen off
+			_ftprintf(stderr, "WARNING: SWT_GTK4 does not support splash screen yet. Therefore it has been disabled. To suppress this message launch with -nosplash\n");
```

Comment:
Would it be possible to support splash-screens with GTK4 or is this too complicated?

Reply:
It will be - this is a temporary state to resolve #831. To do splash screens on GTK4 we need to use GTK4 to start the splash on the native side. That means the launcher will need GTK4 soon.

What I think we will need to do is similar to what we do in SWT. I created #836 for adding the real version of the splash screen for GTK4.

Reply:
PS I was looking for a simple patch to write for eclipse launcher to make sure I understood build flow. If this change seems unnecessary, then we can also just close it and work on the correct GTK4 splash screens at a suitable time in the future.